### ISSUE TITLE
SA1101 shows on 'with' expressions

### ISSUE THREAD

**Issue by CameronAavik** [DotNetAnalyzers/StyleCopAnalyzers#3201]
The following code produces a false positive for SA1101.
It expects you to do `value with { this.Prop = "newValue" }` but that is not valid here.

```csharp
public class Test
{
    public record A
    {
        public string Prop { get; init; }
    }

    public A UpdateA(A value)
    {
        return value with { Prop = "newValue" }; // SA1101: Prefix local calls with this
    }
}
```
